test/config/300-gateway.yaml

```diff
@@ -1,14 +1,38 @@
+# Copyright 2021 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 kind: Gateway
 apiVersion: networking.x-k8s.io/v1alpha1
 metadata:
-  name: simple-gateway
+  name: test-gateway
   namespace: serving-tests
 spec:
   gatewayClassName: istio
-  listeners:  # Use GatewayClass defaults for listener definition.
+  listeners:
   - protocol: HTTP
     port: 80
     routes:
       kind: HTTPRoute
       namespaces:
-        from: "All"
+        from: "Same"
```

Comment:
Won't the HTTProutes be in the namespace of the knative `Service`?

Reply:
Thank you. Yes, `HTTPRoute` should be in the knative service. And `Gateway` should be in the arbitrary namespace. So, I updated to deploy `Gateway` in `knative-serving` (any namespace is fine)  and to use selector to specify `HTTPRoute`.